File: php/src/external/ConversionRateApi.php
<?php

namespace kata\external;


class ConversionRateApi
{

    /**
     * @throws \Exception
     */
    public function getRate(CurrencyIsoCode $source, CurrencyIsoCode $target): float {
        echo 'Getting rate from '.$source->toString().' to '.$target->toString();
        throw new \Exception("401 - Unauthorized, IP not recognized. Are you trying to call us via a unit test?");
    }
}

File: python/external/conversion_rate_api.py
from external.currency_iso_code import CurrencyIsoCode


class ConversionRateApi:

    def get_rate(self, source: CurrencyIsoCode, target: CurrencyIsoCode):
        print("Getting rate from " + source.value + " to " + target.value)
        raise Exception("401 - Unauthorized, IP not recognized. Are you trying to call us via a unit test?")
